bin/happo-ci

```diff
@@ -33,19 +33,21 @@ run-happo() {
   # Run post-install command(s)
   eval "$POST_INSTALL_CMD"
 
-  if npm list happo.io; then
-    npm run --silent happo -- run "$SHA" \
+  if ${NPM_CLIENT} list happo.io; then
+    "$NPM_BIN"/happo run "$SHA" \
     --link "${CHANGE_URL}" \
     --message "${COMMIT_SUBJECT}"
   else
     echo "NPM package happo.io not installed. Assuming this is the first run."
   fi
 }
 
+NPM_BIN=$(${NPM_CLIENT} bin)
+
 # Check if we need to generate a baseline. In some cases, the baseline is
 # already there (some other PR uploaded it), and we can just use the existing
 # one.
-if ! npm run --silent happo -- has-report "$PREVIOUS_SHA"; then
+if ! "$NPM_BIN"/happo has-report "$PREVIOUS_SHA"; then
```

Comment:
i think requiring an npm run-script is a really nice API; this change removes that injection point.

Reply:
In general, I agree. There were a few other reasons we made this change that weren’t listed in the commit message:
- getting rid of unnecessary log output. It can be done with —silent I guess but calling the executable directly seemed more straightforward.
- inconsistencies between the yarn and npm client (which is what led us to always use npm over yarn for these commands). `yarn run` masks exit statuses (always returning 0 or 1). Assuming that both yarn and npm exists seemed unnecessary. 

So I guess it’s down to trade-offs here. Let me think this through for a while. We might be able to handle this with documentation and /or validation/error messages.

Reply:
While the injection is a good thing in general, I haven't yet come across a case where it's being used. So I'm going to go with directly invoking the executable for now. We can always move back to `npm run` if we see a need for it.

Reply:
Even in a yarn repo, using npm seems fine.